### PR TITLE
C# pagestream

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java [hunk 1]
```diff
@@ -633,6 +662,19 @@ private String getTypeName(ProtoElement elem) {
             }
           }
         });
+    if (isPageStreaming) {
+      String[] pageToken = {
+          "/// <param name=\"pageToken\">The token returned from the previous request.",
+          "/// <c>Null</c> or an empty string retrieves the first page.</param>",
+      };
+      String[] pageSize = {
+          "/// <param name=\"pageSize\">The size of page to request, the response will not be larger",
```
Comment (jskeet) [inline src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java]: Nit - I'd either use two sentences, or at least change the comma to a semi-colon.

File: src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java [hunk 2]
```diff
@@ -633,6 +662,19 @@ private String getTypeName(ProtoElement elem) {
...
+          "/// than this, but may be smaller. <c>Null</c> or 0 uses a server-defined page size.</param>",
```
Comment (jskeet) [inline src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java]: I'd change to 

A value of <c>null</c> or 0 uses a [...]

That's easier to read and avoids null having a capital N.

File: src/main/resources/com/google/api/codegen/csharp/wrapper.snip [hunk 3]
```diff
@@ -471,8 +462,22 @@
...
             @end
         @end
     }
 @end
+
+@private pageStreamPartials(pages)
+    // Partial classes to enable page-streaming
+
+    @join page : pages
+        public partial class {@page.requestTypeName} : IPageRequest { }
```
Comment (jskeet) [inline src/main/resources/com/google/api/codegen/csharp/wrapper.snip]: At some point we may need to do some work to make sure we handle requests/responses in a different namespace to the service, but we can punt for now.

**Comment by jskeet** [comment]
A few nits, but generally fine.

File: src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java (hunk 1 above)
Comment (chrisdunelm) [inline src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java]: Done

File: src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java (hunk 2 above)
Comment (chrisdunelm) [inline src/main/java/com/google/api/codegen/csharp/CSharpGapicContext.java]: Done.

File: src/main/resources/com/google/api/codegen/csharp/wrapper.snip (hunk 3 above)
Comment (chrisdunelm) [inline src/main/resources/com/google/api/codegen/csharp/wrapper.snip]: Agreed.

**Pull request merged**